walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,64 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      //Return the times that a meeting can happen today
+      //MeetingRequest holds all the information regarding atendees, and length of meeting
+      //Event has a set of attendees and a specific timerange the atendees are meeting 
+      //TimeRange is a specific set of time a meeting can start and end
+    Collection<TimeRange> slots = new HashSet<TimeRange>();
+    if (request.getDuration() < TimeRange.WHOLE_DAY.duration()) {
+        slots.add(TimeRange.WHOLE_DAY); //Initializing base case
+    }
+
+    for (Event event : events) {
+        for (String attendee : event.getAttendees()) {
+            if (request.getAttendees().contains(attendee)) {
+                Collection<TimeRange> temp = new HashSet<TimeRange>(slots);
+
+                for (TimeRange slot : temp) {
+                    if (slot.overlaps(event.getWhen())) {
+                        //Split slot into slots that dont have overlaps with events
+                        TimeRange range1 = slot.fromStartEnd(slot.start(), event.getWhen().start(), false);
+                        TimeRange range2 = slot.fromStartEnd(event.getWhen().end(), slot.end(), false);
+                        if (slot.contains(event.getWhen())) { 
+                            //inside slot
+                            if (range1.duration() >= request.getDuration()) {
+                                slots.add(range1);
+                            }
+                            if (range2.duration() >= request.getDuration()) {
+                                slots.add(range2);
+                            }
+                        } else if (event.getWhen().contains(slot)) { 
+                            //inside event
```

Comment:
Is something eventually going to happen here? If not, maybe consider an alternate structure for your if/else blocks.